plugins/scaffolder-backend-module-confluence-to-markdowwn/src/actions/confluence/confluenceToMarkdown.test.ts

```diff
@@ -0,0 +1,316 @@
+import { ScmIntegrations } from '@backstage/integration';
+import mockFs from 'mock-fs';
+import os from 'os';
+import fetch from 'node-fetch';
+import type { ActionContext } from '@backstage/plugin-scaffolder-node';
+import { readFile, writeFile } from 'fs-extra';
+import { Readable } from 'stream';
+
+jest.mock('node-fetch');
+jest.mock('fs-extra', () => ({
+  mkdirSync: jest.fn(),
+  readFile: jest.fn().mockResolvedValue('File contents'),
+  writeFile: jest.fn().mockImplementation(() => {
+    return Promise.resolve();
+  }),
+  outputFile: jest.fn(),
+  openSync: jest.fn(),
+  createWriteStream: jest.fn().mockReturnValue(new PassThrough()),
+  writeStream: jest.fn(),
+}));
+const { Response } = jest.requireActual('node-fetch');
+
+describe('transform:confluence-to-markdown', () => {
+  const config = new ConfigReader({
+    confluence: {
+      baseUrl: 'https://nodomain.confluence.com/',
+      token: 'fake_token',
+    },
+  });
+
+  const integrations = ScmIntegrations.fromConfig(
+    new ConfigReader({
+      integrations: {
+        github: [{ host: 'github.com', token: 'token' }],
+      },
+    }),
+  );
+
+  let reader: UrlReader;
+  let mockContext: ActionContext<{
+    confluenceUrls: string[];
+    repoUrl: string;
+  }>;
+
+  const logger = getVoidLogger();
+  jest.spyOn(logger, 'info');
+
+  const mockTmpDir = os.tmpdir();
+
+  beforeEach(() => {
+    reader = {
+      readUrl: jest.fn(),
+      readTree: jest.fn().mockResolvedValue({
+        dir: jest.fn(),
+      }),
+      search: jest.fn(),
+    };
+    mockContext = {
+      input: {
+        confluenceUrls: [
+          'https://nodomain.confluence.com/display/testing/mkdocs',
+        ],
+        repoUrl: 'https://notreal.github.com/space/backstage/mkdocs.yml',
+      },
+      workspacePath: '/tmp',
+      logger,
+      logStream: new PassThrough(),
+      output: jest.fn(),
+      createTemporaryDirectory: jest.fn().mockResolvedValue(mockTmpDir),
+    };
+    mockFs({ [`${mockTmpDir}/src/docs`]: {} });
+  });
+  afterEach(() => {
+    jest.clearAllMocks();
+    mockFs.restore();
+  });
+
+  it('should call confluence to markdown action successfully with results array', async () => {
+    const options = {
+      reader,
+      integrations,
+      config,
+    };
+    const responseBody = {
+      results: [
+        {
+          id: '4444444',
+          type: 'page',
+          title: 'Testing',
+          body: {
+            export_view: {
+              value: '<p>hello world</p>',
+            },
+          },
+        },
+      ],
+    };
+    const responseBodyTwo = {
+      results: [
+        {
+          id: '4444444',
+          type: 'attachment',
+          title: 'testing.pdf',
+          metadata: {
+            mediaType: 'application/pdf',
+          },
+          _links: {
+            download: '/download/attachments/4444444/testing.pdf',
+          },
+        },
+      ],
+    };
+
+    (fetch as jest.MockedFunction<typeof fetch>)
```

Comment:
There's a library called `msw` that can be used in tests to mock server responses. There are several usages of it in this repository for inspiration.

Reply:
fixed